pgmpy/models/BayesianModel.py

```diff
@@ -690,6 +690,32 @@ def fit(self, data, estimator_type=None, state_names=[], complete_samples_only=T
         cpds_list = estimator.get_parameters(**kwargs)
         self.add_cpds(*cpds_list)
 
+    def _predict(self, data, mode):
+        from pgmpy.inference import VariableElimination
+
+        if set(data.columns) == set(self.nodes()):
+            raise ValueError("No variable missing in data. Nothing to predict")
+
+        elif set(data.columns) - set(self.nodes()):
+            raise ValueError("data has variables which are not in the model")
+
+        missing_variables = set(self.nodes()) - set(data.columns)
+        pred_values = defaultdict(list)
+
+        model_inference = VariableElimination(self)
+        for index, data_point in data.iterrows():
+            if mode == 'predict':
```

Comment:
@ankurankan Sorry! I made that commit before reading your comment. Yes! I will revert that.